bobtimus/src/bin/fake_bobtimus.rs

```diff
@@ -26,26 +25,8 @@ async fn main() -> Result<()> {
         usdt_asset_id,
     };
 
-    let bobtimus_filter = warp::any().map({
-        let bobtimus = bobtimus.clone();
-        move || bobtimus.clone()
-    });
-
-    let subscription = bobtimus.rate_service.subscribe();
-    let latest_rate = warp::path!("rate" / "lbtc-lusdt")
-        .and(warp::get())
-        .map(move || routes::latest_rate(subscription.clone()));
-
-    let create_swap = warp::post()
-        .and(warp::path!("swap" / "lbtc-lusdt"))
-        .and(warp::path::end())
-        .and(bobtimus_filter.clone())
-        .and(warp::body::json())
-        .and_then(routes::create_swap);
-
-    warp::serve(latest_rate.or(create_swap))
-        .run(([127, 0, 0, 1], api_port))
-        .await;
+    let routes = http::create_routes(&bobtimus, bobtimus.rate_service.subscribe());
+    warp::serve(routes).run(([127, 0, 0, 1], api_port)).await;
```

Comment:
In theory, you could even move the `warp::serve` into `lib.rs` and expose it as `http::start` :)

This is also fine of course! Given that it is a library interface, it feels a bit weird to take all of this "control" away from the binary. However, one could argue that in this case, we don't want the binaries to have control because they should behave identical when it comes to the API.